core/src/main/scala/org/apache/spark/deploy/rest/StandaloneRestServer.scala

```diff
@@ -139,7 +139,9 @@ private[rest] class StandaloneSubmitRequestServlet(
     val driverExtraLibraryPath = sparkProperties.get("spark.driver.extraLibraryPath")
     val superviseDriver = sparkProperties.get("spark.driver.supervise")
     val appArgs = request.appArgs
-    val environmentVariables = request.environmentVariables
+    // Filter SPARK_LOCAL environment variables from being set on the remote system.
+    val environmentVariables =
+      request.environmentVariables.filterNot(_._1.startsWith("SPARK_LOCAL"))
```

Comment:
You are right, but shouldn't all SPARK_LOCAL* properties be picked up from the local environment of the node where driver is going to be started? 

Not filtering them, would mean, that these local properties are common to all nodes. 

But for this particular bug, `SPARK_LOCAL_DIRS` is not required to be filtered.